src/MIDebugEngine/AD7.Impl/AD7PendingBreakpoint.cs

```diff
@@ -558,6 +561,74 @@ int IDebugPendingBreakpoint2.Virtualize(int fVirtualize)
                 await _bp.EnableAsync(true, _engine.DebuggedProcess);
             }
         }
+
+#if CORECLR
+        /// <summary>
+        /// Get the checksums associated with this pending breakpoint in order to verify the source file
+        /// </summary>
+        /// <param name="hashAlgorithmId">The HashAlgorithmId to use to calculate checksums</param>
+        /// <returns>
+        /// Checksums calculated using the given HashAlgorithmId against the source file represented in this pending breakpoint.
+        /// If no checksums can be calculated, returns null
+        /// </returns>
+        internal Checksum[] GetChecksum(HashAlgorithmId hashAlgorithmId)
+        {
+            IDebugBreakpointChecksumRequest2 checksumRequest = _pBPRequest as IDebugBreakpointChecksumRequest2;
+            if (checksumRequest == null)
+            {
+                return null;
```

Comment:
I would get rid of returning `null` and instead return an empty `IEnumerable`. Also, is there a reason why the return types are `Checksum[]` instead of `IEnumerable<Checksum>` (which is what is consumed in the end?)